aikau/src/main/resources/alfresco/core/BaseWidget.js

```diff
@@ -0,0 +1,120 @@
+ * Provides a base for all Aikau widgets to extend. It is not essential that a widget extends
+ * this module but it can provide significant performance benefits if hundreds or thousands
+ * of instances of the widget need to be rendered at a time. Any widget that does extend this
+ * base must either define a [templateString]{@link module:alfresco/core/BaseWidget#templateString}
+ * or override [createWidgetDom]{@link module:alfresco/core/BaseWidget#createWidgetDom}.
+ * 
+ * @module alfresco/core/BaseWidget
+ * @extends external:dijit/_WidgetBase
+ * @mixes external:dojo/_TemplatedMixin
+ * @mixes module:alfresco/core/Core
+ * @author Dave Draper
+ * @since 1.0.100
+ */
+define(["dojo/_base/declare",
+        "dijit/_WidgetBase", 
+        "dijit/_TemplatedMixin",
+        "alfresco/core/Core"], 
+        function(declare, _WidgetBase, _TemplatedMixin, Core) {
+   
+   return declare([_WidgetBase, _TemplatedMixin, Core], {
+
+      /**
+       * The template that defines the DOM for the widget. If configured this will 
+       * result in the capabilities of dijit/_TemplatedMixin being used to construct the 
+       * DOM. Ideally this should be left as the default value of null and the 
+       * [createWidgetDom]{@link module:alfresco/core/BaseWidget#createWidgetDom} function
+       * will be called - this function should be implemented to use the native capabilities
+       * of the browser to build a DOM for the widget.
+       * 
+       * @instance
+       * @type {string}
+       * @default
+       */
+      templateString: null,
+
+      /**
+       * This extends the function provided by the dijit/_WidgetBase module to prevent any 
+       * attributes from being applied. This changes the default widget behaviour in order
+       * to gain performance improvements. Any parameters that do need to be applied can
+       * be provided by overriding 
+       * [getParametersToApply]{@link module:alfresco/core/BaseWidget#getParametersToApply}.
+       * 
+       * @instance
+       */
+      _applyAttributes: function alfresco_core_BaseWidget___applyAttributes() {
+         var originalParams = this.params;
+         this.params = this.getParametersToApply();
+         this.inherited(arguments);
+         this.params = originalParams;
+      },
+
+      /**
+       * This is called from [_applyAttributes]{@link module:alfresco/core/BaseWidget#_applyAttributes}
+       * to get an object of the parameters that should be applied to the widget. This should only
+       * be implemented as necessary. The more parameters applied the worse the performance at scale.
+       * 
+       * @instance
+       * @overridable
+       * @param {object} originalParams The original parameters provided to the widget.
+       * @return {object} An object of key/value pair parameters to be applied. 
+       */
+      getParametersToApply: function alfresco_core_BaseWidget__getParametersToApply(/*jshint unused:false*/ originalParams) {
+         return null;
```

Comment:
I would say so. Although I also have some improvements for (extended) use of style in mind. When using repeating widgets (i.e. rows / cells / property renderers) it can result in extremely inefficient layout / paint cycles by the browser if we only use inline styling. At PRODYNA I had included dynamic stylesheet handling in our "high performance" base widget to address this.